contracts/staking/GoodFundManager.sol

```diff
@@ -189,76 +237,127 @@ contract GoodFundManager is DAOContract {
-	 * that interest to the reserve contract. Then transfers the given gd which
-	 * received from the reserve contract back to the staking contract and to the
-	 * bridge, which locks the funds and then the GD tokens are been minted to the
-	 * given address on the sidechain
-	 * @param _staking Contract that implements `collectUBIInterest` and transfer iTokeb to
-	 * a given address. The given address should be the same whitelisted `reserve`
-	 * address in the current contract, in case that the given staking contract transfers
-	 * the funds to another contract, zero GD tokens will be minted by the reserve contract.
-	 * Emits `FundsTransferred` event in case which interest has been passed to the `reserve`
-	 */
-	function transferInterest(StakingContract _staking)
-		public
-		reserveHasInitialized
-	//requireDAOContract(address(_staking))
-	{
-		// require(
-		//     canRun(),
-		//     "Need to wait for the next interval"
-		// );
-        address reserveAddress = nameService.addresses(nameService.RESERVE());
-		lastTransferred = block.number.div(blockInterval);
-		ERC20 iToken = ERC20(_staking.iToken());
-		// iToken balance of the reserve contract
-		uint256 currentBalance = iToken.balanceOf(reserveAddress);
-		// collects the interest from the staking contract and transfer it directly to the reserve contract
-		//`collectUBIInterest` returns (iTokengains, tokengains, precission loss, donation ratio)
-	    _staking.collectUBIInterest(reserveAddress);
-
-		// Finds the actual transferred iToken
-		uint256 interest =
-			iToken.balanceOf(reserveAddress).sub(currentBalance);
-		//uint256 effectiveInterest =
-		//	interest.mul(avgEffectiveStakedRatio).div(DECIMAL1e18);
-		//uint256 interestDonated = interest.sub(effectiveInterest);
+    /**
+     * @dev Collects UBI interest in iToken from a given staking contract and transfers
+     * that interest to the reserve contract. Then transfers the given gd which
+     * received from the reserve contract back to the staking contract and to the
+     * bridge, which locks the funds and then the GD tokens are been minted to the
+     * given address on the sidechain
+     */
+    function collectInterest(address[] memory _stakingContracts)
+        public
+        reserveHasInitialized
+        //requireDAOContract(address(_staking))
+    {
+        // require(
+        //     canRun(),
+        //     "Need to wait for the next interval"
+        // );
+        uint initialGas = gasleft();
+        lastTransferred = block.number.div(blockInterval);
+        ERC20 iToken = ERC20(nameService.getAddress("CDAI"));
+        // iToken balance of the reserve contract
+        uint256 currentBalance = iToken.balanceOf(nameService.addresses(nameService.RESERVE()));
+        
+        for(uint i = _stakingContracts.length - 1; i >= 0; i--){ // zxelements are sorted by balances from lowest to highest 
+		
+            if(_stakingContracts[i] != address(0x0)){
```

Comment:
why do we need this if?

Reply:
Because calcSortedContracts function gets whole active contracts then check if it has interest or not or there is enough gas left to collect interest then any of these conditions did not meet it sets this addresses to 0x0 so we can just skip this ones

Reply:
ok